ckan/views/api.py

```diff
@@ -473,12 +473,12 @@ def i18n_js_translations(
     if lang not in get_locales_from_config():
         return _finish_bad_request('Unknown locale: {}'.format(lang))
 
-    ckan_path = os.path.join(os.path.dirname(__file__), u'..')
-    source = os.path.abspath(os.path.join(ckan_path, u'public',
-                             u'base', u'i18n', u'{0}.js'.format(lang)))
+    js_translations_folder = get_js_translations_dir()
+
+    source = os.path.join(js_translations_folder, f"{lang}.js")
     if not os.path.exists(source):
-        return u'{}'
-    translations = json.load(io.open(source, u'r', encoding='utf-8'))
+        return "{}"
+    translations = json.load(io.open(source, "r", encoding="utf-8"))
```

Comment:
I see this existed before this PR, but why are we parsing the json only to immediately send it in the response? We could use LazyJSONObject here or better yet have them served as normal files with nginx or like webassets with x-sendfile.

Reply:
Doesn't need to be part of this PR, I only noticed because of the diligent `u'string'` cleanup here.